pkg/cmd/release-payload-controller/prowjob_controller.go

```diff
@@ -3,54 +3,50 @@ package release_payload_controller
 import (
 	"context"
 	"fmt"
+	releasepayloadclient "github.com/openshift/release-controller/pkg/client/clientset/versioned/typed/release/v1alpha1"
 	releasepayloadinformer "github.com/openshift/release-controller/pkg/client/informers/externalversions/release/v1alpha1"
-	releasepayloadlister "github.com/openshift/release-controller/pkg/client/listers/release/v1alpha1"
 	releasecontroller "github.com/openshift/release-controller/pkg/release-controller"
 	"github.com/openshift/release-controller/pkg/releasepayload/controller"
 	"github.com/openshift/release-controller/pkg/releasepayload/status"
 	"k8s.io/apimachinery/pkg/api/errors"
-	v1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
-	prowjobinformer "k8s.io/test-infra/prow/client/informers/externalversions/prowjobs/v1"
-	prowjoblister "k8s.io/test-infra/prow/client/listers/prowjobs/v1"
-	"time"
-
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog/v2"
+	v1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowjobinformer "k8s.io/test-infra/prow/client/informers/externalversions/prowjobs/v1"
+	prowjoblister "k8s.io/test-infra/prow/client/listers/prowjobs/v1"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
 type ProwJobStatusController struct {
-	releasePayloadNamespace string
-	releasePayloadLister    releasepayloadlister.ReleasePayloadLister
-
-	prowJobLister prowjoblister.ProwJobLister
-
-	eventRecorder events.Recorder
+	*ReleasePayloadController
 
-	cachesToSync []cache.InformerSynced
-
-	queue workqueue.RateLimitingInterface
+	prowJobNamespace string
```

Comment:
where do you use this?

Reply:
I'm working on a downstream PR for the `prowjob_controller`, which monitors for updates to Prowjobs and updates the corresponding ReleasePayload accordingly.